sdk/communication/azure-communication-chat/azure/communication/chat/_shared/token_exchange.py

```diff
@@ -0,0 +1,170 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import json
+from typing import Tuple, Any
+from azure.core.credentials import AccessToken
+from azure.core.pipeline import Pipeline, PipelineResponse
+from azure.core.pipeline.policies import BearerTokenCredentialPolicy
+from azure.core.exceptions import ClientAuthenticationError, HttpResponseError
+from entra_token_guard_policy import EntraTokenGuardPolicy, AsyncEntraTokenGuardPolicy
+from azure.core.credentials_async import AsyncTokenCredential
+from azure.core.credentials import TokenCredential
+from typing import List, Optional
+from datetime import datetime, timezone
+from dateutil import parser as dateutil_parser
+
+from azure.core.pipeline.policies import AsyncBearerTokenCredentialPolicy
+from azure.core.pipeline import AsyncPipeline, PipelineResponse
+
+
+TEAMS_EXTENSION_SCOPE_PREFIX = "https://auth.msft.communication.azure.com/"
+COMMUNICATION_CLIENTS_SCOPE_PREFIX = "https://communication.azure.com/clients/"
+TEAMS_EXTENSION_ENDPOINT = "/access/teamsPhone/:exchangeAccessToken"
+TEAMS_EXTENSION_API_VERSION = "2025-03-02-preview"
+COMMUNICATION_CLIENTS_ENDPOINT = "/access/entra/:exchangeAccessToken"
+COMMUNICATION_CLIENTS_API_VERSION = "2024-04-01-preview"
```

Comment:
Aren't we also using `2025-03-02-preview` for Entra? I thought we had consolidated on one single version

Reply:
fixed everywhere